R/plot_timeline.R

```diff
@@ -2,12 +2,13 @@
 #'
 #' @param data data.frame Should have the same format as output of
 #'   `prep_timeline()` and contain columns: 'direction', 'year',
-#'   'exposure_weighted_net_alignment', 'group_id'.
+#'   'exposure_weighted_net_alignment', by_group.
```

Comment:
I might not entirely understand the logic here but my impression is that if there is a `by_group` column present in the input data this is used for grouping the data in plots, otherwise all data is used and not grouped in any way? In that case, do we need to have it both as a requirement for the data and as a separate argument?

Reply:
not sure I fully understand your suggestion. Do you mean it should be sufficient to use `by_group` in the `prep_*() `functions and then just use whatever "additional" variables are available after `prep_*()` for grouping in `plot_*()`, thereby removing the need to explicitly set `by_group` in any of the `plot_*()` functions?

Reply:
I meant that I don't understand why we require both `by_group` parameter and then also we require that a column called `by_group` is present. Now I understand that `by_group` is actually varying. We should make it clearer in the documentation that by_group is not a column name but a variable. I think you did it in one of the functions.

Reply:
I see, makes sense

Reply:
updated